src/creation.cpp

```diff
@@ -0,0 +1,436 @@
+#include "creation.hpp"
+
+#include <pybind11/detail/common.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <s2/s2latlng.h>
+#include <s2/s2loop.h>
+#include <s2/s2point.h>
+#include <s2/s2polygon.h>
+#include <s2/s2polyline.h>
+#include <s2geography.h>
+#include <s2geography/geography.h>
+
+#include <array>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "geography.hpp"
+#include "pybind11.hpp"
+
+namespace py = pybind11;
+namespace s2geog = s2geography;
+using namespace spherely;
+
+//
+// ---- S2geometry object creation functions.
+//
+
+S2Point make_s2point(double lng, double lat) {
+    return S2LatLng::FromDegrees(lat, lng).ToPoint();
+}
+
+S2Point make_s2point(const std::pair<double, double> &point) {
+    return S2LatLng::FromDegrees(point.second, point.first).ToPoint();
+}
+
+S2Point make_s2point(const Geography *point_ptr) {
+    check_geog_type(*point_ptr, GeographyType::Point);
+    auto s2geog_obj = static_cast<const s2geog::PointGeography &>(point_ptr->geog());
+
+    if (s2geog_obj.Points().empty()) {
+        throw EmptyGeographyException("cannot create s2geometry point from empty POINT Geography");
+        // TODO: what do we want here? NaN or S2Point default constructor? It depends?
+        // return S2Point::NaN();
```

Comment:
Or indeed disallow as you do now? I think I would be fine not enabling the user to create a MultiPoint with empty Point

Reply:
Yes raising an error is good in the context here. This comment was in case we later reuse this function in other contexts (I'll clarify it).